serve/filters/filter/tx.go

```diff
@@ -0,0 +1,203 @@
+package filter
+
+import (
+	"sort"
+
+	"github.com/gnolang/gno/tm2/pkg/bft/types"
+)
+
+// filterPriority defines the priority of a filter condition.
+// The higher the priority, the earlier the condition is applied.
+//
+// This concept is borrowed from operator precedence.
+type filterPriority int
+
+const (
+	HeightPriority filterPriority = iota // highest priority
+	IndexPriority
+	GasUsedPriority
+	GasWantedPriority // lowest priority
+)
+
+type condition struct {
+	filter   func(*types.TxResult) bool
+	priority filterPriority
+}
+
+// TxFilter holds a slice of transaction results.
+// It provides methods to manipulate and query the transactions.
+type TxFilter struct {
+	*baseFilter
+	// txrs represents the transactions in the filter.
+	txrs []*types.TxResult
+	// conditions holds the filtering conditions.
+	conditions []condition
+}
+
+// NewTxFilter creates a new TxFilter object.
+func NewTxFilter() *TxFilter {
+	return &TxFilter{
+		baseFilter: newBaseFilter(TxFilterType),
+		txrs:       make([]*types.TxResult, 0),
+		conditions: make([]condition, 0),
+	}
+}
+
+// GetHashes iterates over all transactions in the filter and returns their hashes.
+//
+// It appends `nil` to the result slice if the transaction or its content is `nil`.
+// This ensures that the length og the returned slice matches the number of transactions in the filter.
+func (tf *TxFilter) GetHashes() [][]byte {
+	tf.Lock()
+	defer tf.Unlock()
+
+	hashes := make([][]byte, 0, len(tf.txrs))
+
+	for _, txr := range tf.txrs {
+		if txr == nil || txr.Tx == nil {
+			hashes = append(hashes, nil)
+
+			continue
+		}
+
+		hashes = append(hashes, txr.Tx.Hash())
+	}
+
+	return hashes
+}
+
+// GetChanges retrieves and returns all the transactions in the filter.
+//
+// It also resets the transactions and prepare the filter for new transactions.
+func (tf *TxFilter) GetChanges() any {
+	tf.Lock()
+	defer tf.Unlock()
+
+	changes := make([]*types.TxResult, len(tf.txrs))
+	copy(changes, tf.txrs)
+
+	tf.txrs = tf.txrs[:0] // reset for new transactions
+
+	return changes
+}
+
+// UpdateWithTx adds a transaction to the filter.
+func (tf *TxFilter) UpdateWithTx(txr *types.TxResult) {
+	tf.Lock()
+	defer tf.Unlock()
+
+	tf.txrs = append(tf.txrs, txr)
+}
+
+// ClearConditions resets the previously set conditions from the filter.
+func (tf *TxFilter) ClearConditions() *TxFilter {
```

Comment:
This method should be removed completely, if we consider that filters are managed top-level (either they exist or they don't, and they can't be modified during their existence)